### PR TITLE
ZenDNN model compiler

### PR DESCRIPTION
**Pull request by LakshmiKumar23**
Introducing nnir_to_zendnn.py script. 

Supports below layers currently:
1. Convolution
2. Pooling
3. ReLU
4. Batch Norm
5. Inner Product/Fully Connected Layer
6. LRN 
7. Sum
8. Softmax

Supported models with this PR:

1. Mnist - Caffe
2. Vgg16 - Caffe
3. Vgg19 - Caffe
4. Renset50 - Caffe
5. Alexnet - Caffe

### CODE REVIEW

**Comment by hansely** [comment]
caffe->nnir->zenDNN conversion works using the model compiler.
However, I wasn't able to build & run the app since my system hangs while building zenDNN.

File: model_compiler/python/nnir_to_zendnn.py [hunk 1]
```diff
@@ -0,0 +1,1024 @@
...
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+
+import os, sys, struct
+import datetime, pytz
+import nnir as ir
+
+def generateLicenseForScript(f):
+    f.write( \
+"""################################################################################
```
Comment (LakshmiKumar23) [inline model_compiler/python/nnir_to_zendnn.py:27]: @kiritigowda Already fixed it and pushed

File: model_compiler/python/nnir_to_zendnn.py [hunk 2]
```diff
@@ -0,0 +1,1024 @@
...
+
+# Bring in weights & images to build folder
+FILE(GLOB IMAGE_LIST ${CMAKE_CURRENT_SOURCE_DIR}/data/images/*.png)
+FILE(GLOB IMAGE_BIN_LIST ${CMAKE_CURRENT_SOURCE_DIR}/data/images/*.bin)
+file(COPY ${CMAKE_CURRENT_SOURCE_DIR}/data/weights.bin DESTINATION ${CMAKE_CURRENT_BINARY_DIR}/data/)
+file(COPY ${IMAGE_LIST} DESTINATION ${CMAKE_CURRENT_BINARY_DIR}/images/)
+file(COPY ${IMAGE_BIN_LIST} DESTINATION ${CMAKE_CURRENT_BINARY_DIR}/images/)
+
+message("-- ${PROJECT_NAME} -- Using Compiler - Path:" ${CMAKE_CXX_COMPILER} "\t
+        Version:" ${CMAKE_CXX_COMPILER_VERSION} "\tCompiler:" ${CMAKE_CXX_COMPILER_ID}${ColourReset})
```
Comment (LakshmiKumar23) [inline model_compiler/python/nnir_to_zendnn.py:191]: @kiritigowda Already fixed it and pushed

File: model_compiler/python/nnir_to_zendnn.py [hunk 3]
```diff
@@ -0,0 +1,1024 @@
...
+// Intialize weights & bias tensors
+static int initializeTensor(std::vector<float> *tensor, size_t tensorSize, FILE *fp, const char *binaryFilename)
+{
+    size_t itemsize = sizeof(float);
+    unsigned int h[2] = {0};
+
+    fread(h, 1, sizeof(h), fp);
+    if (h[0] != 0xf00dd1e1 || (size_t)h[1] != (tensorSize * itemsize))
+    {
+        printf("ERROR: invalid data (magic,size)=(0x%x,%x) in %s at byte position %lu -- expected size is %ld\\n",
```
Comment (LakshmiKumar23) [inline model_compiler/python/nnir_to_zendnn.py:250]: @kiritigowda Already fixed it and pushed

**Review by rrawther — APPROVED** [review]
@hansely to test
